sdk/eventhub/event-hubs/consuming-events.md

```diff
@@ -0,0 +1,85 @@
+# Consuming events
+
+The `EventHubConsumerClient` provides a _push-based_ model for receiving events.
+It exposes a `subscribe` method that lets you specify:
+
+- Whether to read from a specific partition or all partitions.
+- User-provided functions that will be called while your subscription is active.
+- `SubscribeOptions` that let you control how events are received, including batching options.
+
+`subscribe` also returns a `Subscription` object that you can use to stop receiving events or
+to determine if the subscription is actively receiving events.
+
+Calling `subscribe` immediately starts the subscription.
+Once the subscription starts, the following things happen for **each** partition being read.
+
+1.  If supplied, the `partitionInitialize` function is invoked each time the subscription is
+    about to begin reading from a partition.
+    The `PartitionContext` passed to `partitionInitialize` can be used to determine which partition
+    is about to be read from.
+    Events won't be received for the partition indicated until after this function returns if it was provided.
+    Example:
+
+```js
+async processInitialize(context) {
+ console.log(`Initializing partition "${context.partitionId}".`);
+}
+```
+
+2.  The `processEvents` function is invoked with an array of events as they are received.
+    The maximum number of events returned in an array, as well as the maximum amount of time to wait for
+    events to be received are configurable through the `SubscribeOptions` passed to `subscribe`.
```

Comment:
We can use a line break here to separate what gets passed to subscribe vs processEvents.

```suggestion
    events to be received are configurable through the `SubscribeOptions` passed to `subscribe`.
    
```